app/models/provider.rb

```diff
@@ -13,6 +13,8 @@ class Provider < ApplicationRecord
   before_save :update_searchable, if: :accredited_provider?
   before_create :set_defaults
 
+  after_save :update_courses_program_type, if: :provider_type_previously_changed?
```

Comment:
Are there a lot of places where the `provider_type` can change and the `program_type` needs to be updated? Can `update_courses_program_type` method be included in a service/controller/form object when the `provider_type` changes? 

I personally don't like using `after_save` mainly because it's invoked every time you save 😄 and there can be unintended consequences.

If you think the columns need to be in sync at all times then it's fine, just thought I would ask 👍🏻

Reply:
OK, that's fine. I was just not sure if we need an after save but if the developer is changing things in the rails console than it's fine 👍🏻